torch/csrc/api/include/torch/nn/functional/activation.h

```diff
@@ -81,6 +82,95 @@ inline Tensor softplus(const Tensor& input, const SoftplusOptions& options) {
   return torch::softplus(input, options.beta(), options.threshold());
 }
 
+inline Tensor multi_head_attention_forward(
+  const Tensor& query,
+  const Tensor& key,
+  const Tensor& value,
+  int64_t embed_dim_to_check,
+  int64_t num_heads,
+  const Tensor& in_proj_weight,
+  const Tensor& in_proj_bias,
+  const c10::optional<Tensor>& bias_k,
```

Comment:
I think we should also use `const Tensor&` to represent an optional tensor (its null value would be the undefined tensor `Tensor()`), which is consistent with how optional tensor is represented elsewhere in the C++ API (e.g. custom autograd functions).